patches/tModLoader/Terraria/ModLoader/ItemLoader.cs

```diff
@@ -1428,6 +1428,62 @@ public static void OpenVanillaBag(string context, Player player, int arg)
 
 			return item1.ModItem?.CanStackInWorld(item2) ?? true;
 		}
+		
+		private static HookList HookOnStack = AddHook<Action<Item, Item, int, bool>>(g => g.OnStack);
+
+		/// <summary>
+		/// Calls CanStack.  Returns false if CanStack is false.  Calls StackItems if CanStack is true<br/>
+		/// Stacks item1 and item2.  Calls all GlobalItem.OnStack and ModItem.OnStack hooks if item1.stack < item1.maxStack.<br/>
+		/// </summary>
+		/// <param name="item1">Item where the stack is being increased.</param>
+		/// <param name="item2">Item where the stack is being reduced.  If remaining stack is <=0, it is set to defaults.</param>
+		/// <param name="numTransfered">Amount to be transfered </param>
+		/// <param name="stack2">The final stack of item2</param>
+		public static bool TryStackItems(Item item1, Item item2, out int numTransfered, bool reduceItem2Stack = true) {
+			numTransfered = 0;
+			if (!CanStack(item1, item2))
+				return false;
+
+			StackItems(item1, item2, out numTransfered, reduceItem2Stack);
+
+			return true;
+		}
+
+		/// <summary>
+		/// Stacks item1 and item2.  Calls all GlobalItem.OnStack and ModItem.OnStack hooks if item1.stack < item1.maxStack.
+		/// </summary>
+		/// <param name="acceptingItem">Item where the stack is being increased.</param>
+		/// <param name="transferingItem">Item where the stack is being reduced.  If remaining stack is <=0, it is set to defaults.</param>
+		/// <param name="numTransfered">Amount to be transfered </param>
+		/// <param name="stack2">The final stack of item2</param>
+		public static void StackItems(Item acceptingItem, Item transferingItem, out int numTransfered, bool reduceItem2Stack = true, int numToTransfer = int.MinValue, bool firstStackSplit = false) {
+			if (numToTransfer == int.MinValue) {
+				numTransfered = 0;
+				int numToMaxStack = acceptingItem.maxStack - acceptingItem.stack;
+				if (numToMaxStack <= 0)
+					return;
+
+				numTransfered = Math.Min(transferingItem.stack, numToMaxStack);
+			}
+			else {
+				numTransfered = numToTransfer;
+			}
+
+			foreach (var g in HookOnStack.Enumerate(globalItems)) {
+				g.OnStack(acceptingItem, transferingItem, numTransfered, firstStackSplit);
```

Comment:
This order `acceptingItem, transferingItem` is actually the opposite of what the documentation in `OnStack` says. I think the documentation is what people will expect, so you should swap the order _here_

Reply:
I kept it the same order and swapped the documentation so it's consistent with the CanStack() and CanStackInWorld() hooks.